evaluator/src/main/kotlin/SourceTreeResolver.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.evaluator
+
+import java.io.File
+
+import org.apache.logging.log4j.kotlin.Logging
+
+import org.ossreviewtoolkit.downloader.Downloader
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.DownloaderConfiguration
+import org.ossreviewtoolkit.utils.ort.createOrtTempDir
+
+fun interface SourceTreeResolver {
+    companion object : Logging {
+        fun forRemoteRepository(vcsInfo: VcsInfo) =
+            SourceTreeResolver {
+                logger.info {
+                    "Downloading ${vcsInfo.type} source code repository from ${vcsInfo.url}..."
+                }
```

Comment:
The `Downloader` already prints a very similar log:
https://github.com/oss-review-toolkit/ort/blob/2d27a2244f660e43b192058221805721033aa7ce/downloader/src/main/kotlin/Downloader.kt#L187-L189

Reply:
I thought it was good to at least print one log statement when doing something exotic like downloading the sources from the evaluator. If you agree, maybe we can re-phrase the log comment like e.g. 

"The evaluator rules require the project source code. Cloning $url."

What do you think?